node/cn/api_tracer.go

```diff
@@ -769,6 +795,26 @@ func (api *PrivateDebugAPI) traceTx(ctx context.Context, message blockchain.Mess
 	}
 }
 
+// stateAt returns the given block's state from cached stateDB with the GC lock or by regenerating.
+func (api *PrivateDebugAPI) stateAt(block *types.Block, reexec uint64) (*state.StateDB, func(), error) {
+	var stateDB *state.StateDB
+
+	// If we have the state fully available, use that.
+	stateDB, err := api.cn.blockchain.StateAtWithGCLock(block.Root())
+	if err != nil {
+		// If no state is locally available, the desired state will be generated.
```

Comment:
What about defining a function variable here?  e.g.,
```
var f = func() {}

...

return nil, f, err

...

return stateDB, f, nil
```

Reply:
I changed it. :)